File: 0100-same-tree/0100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        que = [[p,q]]
        
        while que:
            curP, curQ = que.pop(0)
            if curP and curQ and curP.val == curQ.val:
                que.append([curP.left, curQ.left])
                que.append([curP.right, curQ.right])
            
            elif curP != curQ:
                return False
        return True
#         dfs. time:o(n), space: o(n)
#         if not p and not q: return True
#         elif not p or not q or p.val != q.val: return False
        
#         return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
              
    

File: 0100-same-tree/0100-same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {
    
    // if(!p && !q) return true;
    // else if(!p || !q ) return false;
    // else if(p.val !== q.val) return false;
    // return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
    
    let queue = [[p,q]];
    while(queue.length){
        let [p,q] = queue.pop();
        if(!q && !p) return true;
        if(!p || !q) return false;
        if(p.val !== q.val ) return false;
        
        if(p.left || q.left) queue.push([p.left, q.left]);
        if(p.right || q.right) queue.push([p.right, q.right]);
    }
    return true
};